tst/EnergyPlus/unit/HVACUnitarySystem.unit.cc

```diff
@@ -228,6 +245,538 @@ class ZoneUnitarySystemTest: public EnergyPlusFixture {
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 1 ), 0.1010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 2 ), 0.2000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 2 ), 0.2010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 3 ), 0.3000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 3 ), 0.3010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 4 ), 0.4000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 4 ), 0.4010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 5 ), 0.5000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 5 ), 0.5010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 6 ), 0.6000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 6 ), 0.6010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 7 ), 0.7000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 7 ), 0.7010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 8 ), 0.8000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 8 ), 0.8010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 9 ), 0.9000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 9 ), 0.9010, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).CoolingVolFlowRatio( 10 ), 1.0000, 0.00001 );
+	EXPECT_NEAR( HVACUnitarySystem::DesignSpecMSHP( 1 ).HeatingVolFlowRatio( 10 ), 1.0000, 0.00001 );
+
+	// autosized air flow and capacity, unitary sytsem capacity matches coils
+	EXPECT_EQ( UnitarySystem( 1 ).MaxCoolAirVolFlow, 1.5 );
+	EXPECT_EQ( UnitarySystem( 1 ).MaxHeatAirVolFlow, 1.5 );
+	EXPECT_NEAR( UnitarySystem( 1 ).DesignCoolingCapacity, 31732.204, 0.001 );
+	EXPECT_EQ( UnitarySystem( 1 ).DesignCoolingCapacity, VarSpeedCoil( 1 ).RatedCapCoolTotal );
+	EXPECT_NEAR( UnitarySystem( 1 ).DesignHeatingCapacity, 31732.204, 0.001 );
+	EXPECT_EQ( UnitarySystem( 1 ).DesignHeatingCapacity, VarSpeedCoil( 2 ).RatedCapHeat );
+	// variable speed coils size air flow differently than other models. The design air volume flow rate is back calculated from design capacity
+	EXPECT_EQ( VarSpeedCoil( 1 ).RatedAirVolFlowRate, VarSpeedCoil( 1 ).RatedCapCoolTotal * VarSpeedCoil( 1 ).MSRatedAirVolFlowPerRatedTotCap( 10 ) );
+	EXPECT_NEAR( VarSpeedCoil( 1 ).RatedAirVolFlowRate, 1.77264, 0.00001 ); // different than unitary system air volume flow rate
+	EXPECT_NEAR( VarSpeedCoil( 2 ).RatedAirVolFlowRate, 1.70, 0.01 ); // VS DX heating coil was not autosized
```

Comment:
These tests highlight how the VS DX coil model air flow sizing is so different from other models. The autosized air flow rate for unitary system = 1.5, the autosized air flow rate for VS DX cooling = 1.772. Not sure we can get to something that resolves the difference because of how the VS DX coil model uses hard-sized inputs for air flow and capacity at each speed to "scale" the autosized data. But we can try to make this better. It's important because Trane is relying on the VS DX coil model for it's GUI.